### ISSUE TITLE
Waring system

### ISSUE THREAD

**Issue by diziara** [MageCity/LongSphinx#94]
It would be cool to have a formalized warning system in the bot. Such as:
`!warn @user reason`
to give us somewhere to keep notes on people violating rules or being a problem.

It would also need a command like 
`!warn view @user` to see what their warn history is.

**Comment by MageCity** [comment]
A few clarifications needed, I think:

1. Permissions: Can I just make it "anyone that has the 'kick members' permission can warn people? Or do I need to configure a role per server?
2. What should the warning history look like? Just a list of `[Date]: [warner] warned for [reason]`?
3. Should warnings be visible at all on servers where they weren't received? Should off-server warnings in the history be shown with some sort of flag on them, or maybe just a `and 3 other off-server warnings` at the end?

**Comment by diziara** [comment]
1. I think it should be able to ban, but if it's not crazy, a setting to choose between kick or ban might be a good idea. 
2. Yeah, it doesn't need to be complicated
3. I'm of mixed thoughts. Maybe a config for if the admin wants to share and/or see other servers data?

**Comment by diziara** [comment]
re 2. The record should maybe include who warned the user. While I hope that it won't matter in the context of where Cecil and Tim are currently deployed, if someone on staff ends up outing themselves as crap, it should be clear if someone's warn history includes entries from that person, so that it can be audited if necessary. 

Which leads to also: should it be possible for staff/admin to be able to strike someone's warning from their history if it comes to light that a warning was unjustified?

**Comment by WaltWh** [comment]
That's what the "warner" part was meant to be. I was thinking that would be something like:

`/warnlog <user>`

> Warnings received by <user>
> 2021/01/22: Death Kitten warned for loud chewing
> 2021/03/14: Tin Wizard warned for insulting robots
> And 3 public warnings on other servers, most recent on 2022/01/07


And I can absolutely make it a DeletableListView like the other ones I'm using in the new system. Should removing a warning take different permissions than adding one?

**Comment by diziara** [comment]
Yeah. Like, if the warnings are allowed to be set by people with kick perms as well as ban perms, then they should only be removable by people with ban perms. And I know I said earlier that warning should be ban perms only, but upon reflection, allowing people with kick perms to be able to set warnings is probably good, but only allowing someone with ban perms to delete them would reduce abuse.

**Comment by diziara** [comment]
> 2021/01/22: warned by Death Kitten for loud chewing
Bitch eating crackers :joy_cat:

**Comment by WaltWh** [comment]
I'm integrating the warnlogs with the bans, kicks, and timeouts; what should the minimum permission be to view the whole discipline log?

**Comment by diziara** [comment]
I'm torn between limiting it to those with ban perms, but being able to read the list when you are able to add to it doesn't seem out of order either.

**Comment by diziara** [comment]
If those below ban can read it, it should be only ban or admin that can remove entries
